src/tpm2.c

```diff
@@ -5648,6 +5665,69 @@ UINT16 TPM2_GetVendorID(void)
     return vid;
 }
 
+/* Stores nameAlg + the digest of nvPublic in buffer, total size in size */
+int TPM2_HashNvPublic(TPMS_NV_PUBLIC* nvPublic, byte* buffer, UINT16* size)
+{
+    int rc;
+#ifndef WOLFTPM2_NO_WOLFCRYPT
+    int hashSize, nameAlgSize;
+    UINT16 nameAlgValue;
+    wc_HashAlg hash;
+    enum wc_HashType hashType;
+    byte appending[sizeof(TPMS_NV_PUBLIC)];
+    TPM2B_DATA digest;
```

Comment:
This is an oversized buffer and will use more stack than required. Either use the correctly sized buffer `TPMU_NAME`, which is max digest + 2 bytes (handle) or use the incoming 'buffer` directly and remove this.

Reply:
I went with using directly buffer[2] , thanks.